### ISSUE TITLE
Combination of rules

### ISSUE THREAD

**Issue by akond** [AbhinavOmprakash/luna#2]
I was trying to do 
```
(luna/pre
  [:match-enc
    [:match :digits :exactly 2 :times]
    [:match ":"]
    [:match :digits :exactly 2 :times]])
```
and got  `#""`

**Comment by AbhinavOmprakash** [comment]
ah, the syntax to do that is this. 

```clojure 
(pre [:match-enc [:digits :exactly 2 :times ":" :digits :exactly 2 :times]]) 
; #"(?:\d{2}:\d{2})"
```
`match-enc` is a sort of substitue for `match` 
so you can do this as well and it is valid. 

```clojure
(pre [:match [:digits :exactly 2 :times ":" :digits :exactly 2 :times]])  
; #"\d{2}:\d{2}"
```

**Comment by akond** [comment]
Let me rephrase the problem. Here is my `net.cgrand.regex` code:

```
(def year (cr/repeat cr/digit 4))
(def month (cr/repeat cr/digit 2))
(def day (cr/repeat cr/digit 2))
(def hours (cr/repeat cr/digit 2))
(def mins (cr/repeat cr/digit 2))
(def expr-secs (cr/repeat cr/digit 2))
(def expr-dash \-)
(def expr-date [year expr-dash month expr-dash day])
(def expr-time [hours \: mins (cr/? \: expr-secs)])
(def date+time [expr-date (cr/? cr/space (cr/? expr-time))])
(def log-line [[date+time :as :date] (cr/* cr/space) [(cr/* cr/any) :as :text]])
```
`date+time` and `log-line` are just combinations of other rules. Can I combine rules as easily with lune?

**Comment by AbhinavOmprakash** [comment]
Oh I see. Now your question makes sense. 

I just pushed some commits that lets you combine the rules (haven't made a release yet, would like your input on it).

The changes let you combine the rules like this 

```clojure
(def digits-4 [:match [:digits :exactly 4 :times]])
(def alphabets-5 [:match [:alpha :exactly 5 :times]])

(pre digits-4) ; #"\d{4}"
(pre alphabets-5) ; #"[a-zA-Z]{5}"

(pre [:match [digits-4 "-" alphabets-5]]) ; #"\d{4}-[a-zA-Z]{5}"
(pre [:match [digits-4 "-" alphabets-5 "1234" alphabets-5]]) ; #"\d{4}-[a-zA-Z]{5}1234[a-zA-Z]{5}"
```

**Comment by akond** [comment]
This looks so much better.

**Comment by AbhinavOmprakash** [comment]
Thanks for bringing this up, it has helped improve this library :)
